### DISCUSSION TITLE
Adjusting Executioner parameters during runtime or setting a Stopping criterion

### DISCUSSION THREAD

**Discussion by ttruster** [idaholab/moose#19112] [Q&A Getting Started] [answered]
I am just starting into using MOOSE, so I apologize if this is covered in a test input somewhere, but I haven't located one yet. The analysis I am considering is of a multi element/material model, but I'll describe a simple problem by analogy.
I am desiring to simulation a Maxwell material (spring and dashpot in series) under an applied force that is initially ramped quickly and then helped fixed for an indefinite time (solid mechanics folks: think of a creep experiment). Since the material is nonlinear, I'd like to use the IterationAdaptiveDT TimeStepper for Transient Execution. Thus, the simulation can have a small dt initially, and then adaptively increase once the load is at its hold period. Then, I'd like the simulation to finish/stop once the total elongation/displacement of the material has reached a "criteria"; for instance a certain target value. While I could set the end_time to some value that I think the device/material will pass/exceed the criterion, that time will change with different material coefficients and is thus "hard" to estimate.
I am looking for a good example that talks about the Controls systems like TimePeriods or perhaps one that discusses the control_tags input parameter, which could shed light on how to build such a criteria.

Thank you,

Tim

**Comment by GiudGiud** [comment, marked as answer]
Hi Tim

You can use the steady state detection option, explained here
https://mooseframework.inl.gov/source/executioners/Transient.html. It will terminate the transient in time based on the residual

For a custom termination criteria, you can also use a Terminator https://mooseframework.inl.gov/source/userobject/Terminator.html
You ll have to use postprocessors to retrieve terms in the criteria you want, then a formula can be parsed by the Terminator in the `expression` field. I think this will crash the simulation on request though! Might not be the end you want

Guillaume

**Reply by friedmud** [reply]
`Terminator` is the right way to do this.  The default for the Terminator is to stop the solve at the end of the Timestep (`faile_mode = HARD`) and if you set `error_level = INFO` it will simply print the `message` and then the solve will end at the end of the timestep.

That `error_level` really should be the default... but it's not clear to me that it is.

@andrsd - do you want to weigh in here?

**Reply by ttruster** [reply]
Hi Guillaume and Derek:
I was able to add Terminator to my input file and have the simulation stop using that command with the HARD option. Thanks for the suggestions!
Tim

**Reply by andrsd** [reply]
@friedmud If you are pinging me about the `error_level`, you have the wrong guy - that was done by @dschwen in `84d87ee346f`.

Otherwise, `Terminator` is the way to do this...
